File: Seccion 11 - Funciones en C++/Seccion_11_Dart/27_ejercicio_21.dart

import 'dart:io';

int potencia(int x, int y) {
  if (y == 0) {
    return 1;
  } else if (y == 1){
    return x;
  } else {
    return x * potencia(x, y - 1);
  }
}

void main() {

  int base, exponente;

  stdout.writeln('Digite la base: ');
  base = int.parse(stdin.readLineSync());
  stdout.writeln('Digite el exponente: ');
  exponente = int.parse(stdin.readLineSync());

  print('\nLa potencia de $base^$exponente es: ${potencia(base, exponente)}');
}

File: Seccion 11 - Funciones en C++/Seccion_11_Python/27_ejercicio_21.py

import typing

base: int = 0
exponente: int = 0

def potencia(x: int, y: int) -> int:
    if y == 0:
        return 1
    elif y == 1:
        return x
    else:
        return x * potencia(x, y - 1)

base = int(input('Digite la base: '))
exponente = int(input('Digite el exponente: '))

print('\nLa potencia de {}^{} es: {}'.format(base, exponente, potencia(base, exponente)))